optimum/habana/transformers/generation/utils.py

```diff
@@ -550,19 +550,27 @@ def generate(
                 inputs_tensor, generation_config.pad_token_id, generation_config.eos_token_id
             )
 
-        is_greedy_or_beam_and_bucket = generation_config.bucket_size > 0 and (
-            self._get_generation_mode(generation_config, assistant_model) == GenerationMode.GREEDY_SEARCH
-            or self._get_generation_mode(generation_config, assistant_model) == GenerationMode.BEAM_SEARCH
+        is_greedy_or_beam_and_bucket = (
+            not generation_config.bucket_internal
+            and generation_config.bucket_size > 0
+            and (
+                self._get_generation_mode(generation_config, assistant_model) == GenerationMode.GREEDY_SEARCH
+                or self._get_generation_mode(generation_config, assistant_model) == GenerationMode.BEAM_SEARCH
+            )
         )
         model_kwargs["bucket_size"] = generation_config.bucket_size if generation_config.static_shapes else -1
+        model_kwargs["bucket_internal"] = generation_config.bucket_internal
         model_kwargs["reduce_recompile"] = (
             generation_config.reduce_recompile if generation_config.reduce_recompile is not None else False
         )
         if model_kwargs["reduce_recompile"]:
             assert generation_config.bucket_size
         if generation_config.reuse_cache:
             assert self.config.model_type in ["llama"], "reuse_cache only supported by llama at the moment"
-            assert generation_config.bucket_size <= 0, "reuse_cache and bucketing flags set together"
+            if not generation_config.bucket_internal:
+                assert generation_config.bucket_size <= 0, "reuse_cache and bucketing flags set together"
+            else:
+                assert generation_config.bucket_size >= 0, "bucket_internal and bucket_size flags set together"
```

Comment:
Here, we are in the case where `generation_config.bucket_internal` is True, so if this assert fails (i.e. `generation_config.bucket_size` < 0), it means that bucket_size is not set right? But the error message says otherwise

Reply:
@puneeshkhanna I see you've corrected some error messages. I hope this update won't cause conflict.

Reply:
@xt574chen - I will update my PR once this gets merged first